File: 8.Best-Practice-e-trucchi-avanzati/objc/CM_HandOff/CM_HandOff WatchKit Extension/InterfaceController.m
//
//  InterfaceController.m
//  CM_HandOff WatchKit Extension
//
//  Created by WhiteTiger "sgama la rete" on 28/02/16.
//  Copyright © 2016 WhiteTiger "sgama la rete". All rights reserved.
//

#import "InterfaceController.h"


@interface InterfaceController()

@end


@implementation InterfaceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];

    [self invalidateUserActivity];
}

- (void)willActivate {
    [super willActivate];
    
    [self invalidateUserActivity];
}

- (void)didDeactivate {
    [super didDeactivate];
    
    [self invalidateUserActivity];
}

#pragma mark - IBAction Methods

- (IBAction)actionOption1:(id)sender {
    [self updateUserActivity:@"it.book.handoff-active" userInfo:@{@"option" : @"opzione 1"} webpageURL:nil];
}

- (IBAction)actionOption2:(id)sender {
    [self updateUserActivity:@"it.book.handoff-active" userInfo:@{@"option" : @"opzione 1"} webpageURL:nil];
}

- (IBAction)actionOption3:(id)sender {
    [self updateUserActivity:@"it.book.handoff-active" userInfo:@{@"option" : @"opzione 1"} webpageURL:nil];
}

@end

File: 8.Best-Practice-e-trucchi-avanzati/swift/CM_HandOff/CM_HandOff WatchKit Extension/InterfaceController.swift
//
//  Copyright © 2015 WhiteTiger "sgama la rete". All rights reserved.
//

import WatchKit
import Foundation

class InterfaceController: WKInterfaceController {

    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
        
        invalidateUserActivity()
    }

    override func willActivate() {
        super.willActivate()
        
        invalidateUserActivity()
    }

    override func didDeactivate() {
        super.didDeactivate()
        
        invalidateUserActivity()
    }

    // MARK: - IBAction Methods
    
    @IBAction func actionOption1() {
        updateUserActivity("it.book.handoff-active", userInfo: ["option" : "opzione 1"], webpageURL: nil)
    }
    
    @IBAction func actionOption2() {
        updateUserActivity("it.book.handoff-active", userInfo: ["option" : "opzione 2"], webpageURL: nil)
    }
    
    @IBAction func actionOption3() {
        updateUserActivity("it.book.handoff-active", userInfo: ["option" : "opzione 3"], webpageURL: nil)
    }
}
